lib/alice.ex

```diff
@@ -26,10 +26,19 @@ defmodule Alice do
   end
 
   @doc """
-  Stops an Alice bot instance
+  Stops an Alice bot instance along with its handlers and adapters
   """
   def stop_bot(pid) do
-    Supervisor.terminate_child(Alice.Bot.Supervisor, pid)
+    if Process.alive?(pid) do
+      %{adapters: adapters, handlers: handlers} = :sys.get_state(pid)
+      for adapter <- adapters do
+        Supervisor.terminate_child(Alice.Adapter.Supervisor, Process.whereis(adapter))
```

Comment:
Could `Supervisor.stop` ([docs here](https://hexdocs.pm/elixir/Supervisor.html#stop/3)) with `Alice.Adapter.Supervisor`be used to avoid the loop?

Reply:
I'll look into it, that would be much better. This is what I had to do to get rid of that error messages after the tests run, and I'm all for simplifying it.